File: cpp/1996.the-number-of-weak-characters-in-the-game.cpp
/*
 * @lc app=leetcode id=1996 lang=cpp
 *
 * [1996] The Number of Weak Characters in the Game
 */
#include <vector>
#include <algorithm>
using namespace std;
// @lc code=start
// t:O(Nlog(N)), s:O(1)
class Solution
{
public:
    int numberOfWeakCharacters(vector<vector<int>> &properties)
    {
        sort(properties.begin(), properties.end(), [](vector<int> &l, vector<int> &r)
             { return l[0] == r[0] ? l[1] < r[1] : l[0] > r[0]; });
        int current_max = 0;
        int res = 0;
        for (auto &p : properties)
        {
            if (p[1] < current_max)
            {
                ++res;
            }
            else
            {
                current_max = p[1];
            }
        }
        return res;
    }
};
// @lc code=end

File: py/1996.the-number-of-weak-characters-in-the-game.py
#
# @lc app=leetcode id=1996 lang=python3
#
# [1996] The Number of Weak Characters in the Game
#
from typing import *
# @lc code=start


class Solution:
    def numberOfWeakCharacters(self, properties: List[List[int]]) -> int:
        properties.sort(key=lambda v: [-v[0], v[1]])
        current_max = 0
        res = 0
        for p in properties:
            if p[1] < current_max:
                res += 1
            else:
                current_max = p[1]
        return res
# @lc code=end
